### PR TITLE
Create Blk node for struct vararg

### PR DESCRIPTION
**Pull request by CarolEidt**
When morphing a reference to a struct parameter in a varargs method, it must be a blk node if it is the destination of an assignment.

### CODE REVIEW

**Comment by CarolEidt** [comment]
A better fix might have been to pass the tree node in to the `fgMorphStackArgForVarArgs()` method, but that would have been a slightly more impactful code change.

**Comment by CarolEidt** [comment]
@pgavlin @briansull PTAL
cc @dotnet/jit-contrib

File: src/jit/morph.cpp [hunk 1]
```diff
@@ -10579,7 +10599,7 @@ GenTreePtr Compiler::fgMorphSmpOp(GenTreePtr tree, MorphAddrContext* mac)
                 {
                     // We should check whether op2 should be assigned to a SIMD field or not.
                     // If it is, we should tranlate the tree to simd intrinsic.
-                    assert((tree->gtDebugFlags & GTF_DEBUG_NODE_MORPHED) == 0);
+                    assert(!fgGlobalMorph || ((tree->gtDebugFlags & GTF_DEBUG_NODE_MORPHED) == 0));
```
Comment (briansull) [inline src/jit/morph.cpp]: This fixes the issue that I saw

**Comment by briansull** [comment]
Horrors there are formatting errors:

Failed - Windows_NT x64 Formatting 

How you you see what the error is with this task?
Must you run it locally again to see what it doesn't like?

**Comment by CarolEidt** [comment]
@dotnet-bot test this please

**Comment by JosephTremoulet** [comment]
> How you you see what the error is with this task?
> Must you run it locally again to see what it doesn't like?

If you click the [Details](https://ci.dot.net/job/dotnet_coreclr/job/master/job/x64_windows_nt_formatting_prtest/677/) link of the failed test, it says at the bottom that formatting failed and you can click [Indication 1](https://ci.dot.net/job/dotnet_coreclr/job/master/job/x64_windows_nt_formatting_prtest/677/consoleFull#-452579719e0cf1bf6-b9e4-4679-b910-ada9d1ab2bcb) to go to the spot in the test's log where it shows you what change it wants you to make.  In this case it looks like so:

> ```
> clang-format: there are formatting errors in D:\j\workspace\x64_windows_n---616f30f2\src\jit\morph.cpp
> 01:09:10 At Line 8364 Before:
> 01:09:10             }
> 01:09:10 }
> 01:09:10 After:
> 01:09:10             }
> 01:09:10         }
> ```

You can also then scroll down to the bottom and find the instructions on how to fix it:

> ```
> 01:09:10 There were errors in formatting. Please run jit-format locally with: 
> 01:09:10 
> 01:09:10 	jit-format -a x64 -b Checked -o Windows_NT -c <absolute-path-to-coreclr> --verbose --fix --projects dll
> 01:09:10 
> 01:09:10 
> 01:09:10 Or download and apply generated patch:
> 01:09:10 wget https://ci.dot.net/job/dotnet_coreclr/job/master/job/x64_windows_nt_formatting_prtest/677/artifact/format.patch
> 01:09:10 git apply format.patch
> 01:09:10 
> ```

The hyperlinked file there is another way to view the diff, in this case it looked like [this](https://gist.github.com/JosephTremoulet/97210fd9201d7bc62d1fc25bca4b35e6).

**Comment by CarolEidt** [comment]
@dotnet/jit-contrib ping

**Comment by pgavlin** [comment]
LGTM. Thanks for fixing this!

**Pull request merged**